backend/adapter_processor/exceptions.py

```diff
@@ -64,6 +75,8 @@ def __init__(
         adapter_name: str = "adapter",
     ):
         if detail is None:
+            if adapter_name != "adapter":
```

Comment:
What this conditions makes here?

Reply:
Can we do like this?
            detail = (
                f"Cannot delete '{adapter_name}'. "
                "It is used in a workflow or a prompt studio project"
            )

Reply:
@muhammad-ali-e in the event that we call this exception class and we don't pass an `adapter_name` - the error message would read as `Cannot delete 'adapter'. It is used in a workflow or a prompt studio project.`
I didn't want to single quote the string `adapter` since its a generic placeholder